### PR TITLE
Feature/sensor service

### PR DESCRIPTION
**Pull request by ctacke**

### CODE REVIEW

File: source/Meadow.Core/ThreadedPollingSensorMonitor.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                        {
+                            monitor.ResultProperty = task.GetType().GetProperty("Result");
+                        }
+
+                        var value = monitor.ResultProperty.GetValue(task);
+
+                        // raise an event - not ideal as all sensors get events for all other sensors
+                        // fixing this requires eitehr exposing a "set" method, which I'd prefer be kept internal
+                        // or using reflection to find a set method, which is fragile
+                        SampleAvailable?.Invoke(monitor.Sensor, value);
```
Comment (ctacke) [inline source/Meadow.Core/ThreadedPollingSensorMonitor.cs:101]: @adrianstevens this is the biggest 🤮 in here - open to anything better

File: source/Meadow.Core/SensorService.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            _pollMonitor = new ThreadedPollingSensorMonitor();
+        }
+
+        if (sensor is IPollingSensor s)
+        {
+            s.SensorMonitor?.StopSampling(s);
+
+            s.StopUpdating();
+
+            Resolver.Log.Info($"Replacing monitor on {s.GetType().Name}");
```
Comment (adrianstevens) [inline source/Meadow.Core/SensorService.cs:26]: elevate log type from Info

Comment (adrianstevens) [reply source/Meadow.Core/SensorService.cs:26]: or delete :)

File: source/Meadow.Core/ThreadedPollingSensorMonitor.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            existing.NextReading = sensor.UpdateInterval;
+            existing.EnableReading = true;
+        }
+    }
+
+    private async void SamplingThreadProc()
+    {
+        while (!_stopRequested)
+        {
+            foreach (var monitor in _monitorList)
```
Comment (adrianstevens) [inline source/Meadow.Core/ThreadedPollingSensorMonitor.cs:72]: rename to meta / metalist

**Review by adrianstevens — APPROVED** [review]

**Pull request merged**